shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h

```diff
@@ -39,6 +39,16 @@ typedef NS_ENUM(NSInteger, FlutterKeyboardMode) {
 @property(nonatomic, readonly) BOOL isPresentingViewController;
 @property(nonatomic, readonly) BOOL isVoiceOverRunning;
 @property(nonatomic, retain) FlutterKeyboardManager* keyboardManager;
+
+/**
+ * @brief Whether the status bar is preferred hidden.
+ *
+ *        The |UIViewController:prefersStatusBarHidden| of this ViewController is overriden and
+ * returns `flutterPrefersStatusBarHidden`. This is ignored when
+ * `UIViewControllerBasedStatusBarAppearance` in info.plist of the app project is `false`.
+ */
+@property(nonatomic, assign) BOOL flutterPrefersStatusBarHidden;
```

Comment:
Looking at the structure again here... couldn't you just redeclare `prefersStatusBarHidden` as read-write, to avoid having two getters that do exactly the same thing?

Reply:
Done. I still kept the local var's name to be _flutterPrefersStatusBarHidden so it is clearer that the getter and setter are overridden.